egs/wsj/s5/utils/combine_ali_dirs.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+# Copyright 2016  Xiaohui Zhang  Apache 2.0.
+
+# This script operates on alignment directories, such as exp/tri4a_ali
+
+# Begin configuration section. 
+cmd=run.pl
+extra_files= # specify addtional files in 'src-data-dir' to merge, ex. "file1 file2 ..."
+ref_data_dir= # data directory to be used as reference for rearranging alignments
+num_jobs= # number of jobs used to split the reference data directory.
+# End configuration section.
+set -x
+echo "$0 $@"  # Print the command line for logging
+
+if [ -f path.sh ]; then . ./path.sh; fi
+. parse_options.sh || exit 1;
+
+if [[ $# -lt 2 || ! "$ref_data_dir" || ! "$num_jobs" ]]; then
+  echo "Usage: combine_ali_dirs.sh --ref-data-dir data/train --num-jobs 32 [--extra-files 'file1 file2'] <dest-ali-dir> <src-ali-dir1> <src-ali-dir2> ..."
+  echo "Note, files that don't appear in the first source dir will not be added even if they appear in later ones."
+  echo "Other than alignments, only files from the first src dir are copied."
+  exit 1
+fi
+
+dest=$1;
+shift;
+first_src=$1;
+
+rm -r $dest 2>/dev/null
+mkdir -p $dest;
+
+export LC_ALL=C
+
+for dir in $*; do
+  if [ ! -f $dir/ali.1.gz ]; then
+    echo "$0: check if alignments (ali.*.gz) are present in $dir."
+    exit 1;
+  fi
+done
+
+for dir in $*; do
+  for f in tree; do
+    diff $first_src/$f $dir/$f 1>/dev/null 2>&1
+    if [ $? -ne 0 ]; then
+      echo "$0: Cannot combine alignment directories with different $f files."
+    fi
+  done
+done
+
+for f in final.mdl tree cmvn_opts splice_opts num_jobs $extra_files; do
+  if [ ! -f $first_src/$f ]; then
```

Comment:
not all of these files are required in all recipes, e.g. I'm not sure that splice_opts is required in nnet3.  I would just assume this whole list is optional, and add in whatever you can think of here that we might want to copy over, including final.mat and subsampling_factor [just off the top of my head.]